### ISSUE TITLE
Segfault in load_icons with invalid zip

### ISSUE THREAD

**Issue by Helloman892** [astronautlevel2/Anemone3DS#219]
I won't claim to be familiar with our decompression/icon loading code, so I'll do the best I can.

With an invalid zip (of some persuasion) is in the `themes` directory, scrolling around for a bit will cause a segfault:
```
0x00100000 in _start ()
(gdb) c
Continuing.
Invalid zip being opened
total: 36
extended load
Invalid zip being opened
total: 5
small load
starting thread
Invalid zip being opened
[New Thread 269]
[New Thread 259]
[New Thread 264]
[New Thread 266]

Thread 2 received signal SIGSEGV, Segmentation fault.
[Switching to Thread 269]
load_icons (mutex=1540127, current_list=0xfffffffc) at /home/dylan/git/Anemone3DS/source/loading.c:409
409             C3D_TexDelete(image->tex);
(gdb) bt
#0  load_icons (mutex=1540127, current_list=0xfffffffc) at /home/dylan/git/Anemone3DS/source/loading.c:409
#1  load_icons_thread (void_arg=0x80be408) at /home/dylan/git/Anemone3DS/source/loading.c:438
#2  0x0013e134 in _thread_begin (arg=<optimized out>) at /home/dylan/git/libctru/libctru/source/thread.c:37
#3  0x00000000 in ?? ()
Backtrace stopped: previous frame identical to this frame (corrupt stack?)
```

I suspect that [this zip](https://github.com/astronautlevel2/Anemone3DS/files/4771424/busted_theme.zip)'s existence is ~~due to an error on my part whilst modifying `remote.c#http_get`, but it's an issue nonetheless~~ EDIT: nope, turns out I just had a full SD when downloading. Upgrading this to 'we should definitely handle this' status. You'll need to have enough themes on your console to get it off screen.

**Comment by astronautlevel2** [comment]
Entries seem to be created in the entry list w/o checking the validity of the zip file or if info.smdh exists in directories, and these entries obviously don't have valid icon data, and it seems that when attempting to deallocate this invalid data Anemone crashes. Should be fixed by checking for the existence of info.smdh when creating the entry list - need to do a bit more testing first to be sure, though.

**Comment by Helloman892** [comment]
Fixed in a4532b0f3dc294245305e2af46b356de97574050.
